crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,363 @@
+    }
+    let Some(parent) = JsLogicalExpression::cast(parent) else {
+        return false;
+    };
+    let Some(ope) = parent.operator_token().ok() else {
+        return false;
+    };
+    ope.kind() == operator
+}
+
+fn comparing_length_exp(
+    binary_exp: &JsBinaryExpression,
+    expect_error: &ErrorType,
+) -> Option<AnyJsExpression> {
+    let left = binary_exp.left().ok()?;
+    let operator = binary_exp.operator_token().ok()?.kind();
+    let right = binary_exp.right().ok()?;
+    // Check only when the number appears on the right side according to the original rules.
+    // We assume that you have already complied with useExplicitLengthCheck
+    comparing_length(&left, operator, &right, expect_error)
+}
+
+fn comparing_length(
+    compare_exp: &AnyJsExpression,
+    operator: JsSyntaxKind,
+    value_exp: &AnyJsExpression,
+    expect_error: &ErrorType,
+) -> Option<AnyJsExpression> {
+    let AnyJsExpression::JsStaticMemberExpression(member_exp) = compare_exp else {
+        return None;
+    };
+    let target = member_exp.object().ok()?;
+    let member = member_exp.member().ok()?;
+    if member.text() != "length" || member_exp.is_optional_chain() {
+        return None;
+    }
+    let AnyJsExpression::AnyJsLiteralExpression(AnyJsLiteralExpression::JsNumberLiteralExpression(
+        literal,
+    )) = value_exp
+    else {
+        return None;
+    };
+    let number = literal.as_number()?.round() as i64;
+    // .length === 0
+    if matches!(expect_error, ErrorType::UselessLengthCheckWithEvery)
+        && literal.to_string().trim() == "0"
+        && (operator == T![===] || operator == T![<])
+    {
+        return Some(target);
+    }
+    // .length !== 0
+    if matches!(expect_error, ErrorType::UselessLengthCheckWithSome)
+        && (literal.to_string().trim() == "0" && (operator == T![!==] || operator == T![>])
+            || number > 0 && operator == T![===]
+            || literal.to_string().trim() == "1" && operator == T![>=])
+    {
+        return Some(target);
+    }
+    None
+}
+
+pub type Replacer = (JsLogicalExpression, AnyJsExpression, TextRange);
+
+fn search_logical_exp(
+    any_exp: &AnyJsExpression,
+    replacer: Option<Replacer>,
+    expect_error: &ErrorType,
+    comparing_zeros: &mut HashMap<String, Vec<Replacer>>,
+    array_tokens_used_api: &mut HashSet<String>,
+) -> Option<()> {
+    match any_exp {
+        // || or &&
+        AnyJsExpression::JsLogicalExpression(logical_exp) => {
+            let operator = match expect_error {
+                ErrorType::UselessLengthCheckWithEvery => T![||],
+                ErrorType::UselessLengthCheckWithSome => T![&&],
+            };
+            if logical_exp.operator_token().ok()?.kind() != operator {
+                return None;
+            };
+            let left = logical_exp.left().ok()?;
+            let left_replacer = (logical_exp.clone(), logical_exp.right().ok()?, left.range());
+            search_logical_exp(
```

Comment:
This is performing a complete search within the range of descendant logical expressions. While using a stack instead of a recursive function comes to mind, I believe it would only complicate things further.